src/webgpu/web_platform/canvas/getCurrentTexture.spec.ts

```diff
@@ -179,21 +210,7 @@ g.test('multiple_frames')
         prevTexture = currentTexture;
 
         if (frameCount++ < 5) {
-          // Which method will be used to begin a new "frame"?
-          switch (canvasType) {
-            case 'onscreen':
-              requestAnimationFrame(frameCheck);
-              break;
-            case 'offscreen': {
-              (ctx.canvas as OffscreenCanvas).transferToImageBitmap();
-              // The beginning of frameCheck runs immediately (in the same task), so this
-              // verifies the state has changed synchronously.
-              void frameCheck();
-              break;
-            }
-            default:
-              unreachable();
-          }
+          t.requestNewFrameOnCanvasType(canvasType, ctx, frameCheck);
```

Comment:
[2] I'd remove this abstraction and restore the old code, but with a few changes